common/persistence/cassandraHistoryPersistence.go

```diff
@@ -51,6 +51,13 @@ const (
 		`WHERE domain_id = ? ` +
 		`AND workflow_id = ? ` +
 		`AND run_id = ? `
+
+	templateDeleteWorkflowExecutionPartialHistory = `DELETE FROM events ` +
+		`WHERE domain_id = ? ` +
+		`AND workflow_id = ? ` +
+		`AND run_id = ? ` +
+		`AND first_event_id >= ? ` +
+		`AND first_event_id < ?`
```

Comment:
Lets not allow the user to pass upper bound for delete.  Current use case is to delete the entire tail, so we should only allow callers to pass in the start index.